ddtrace/internal/datadog/profiling/stack_v2/src/stack_renderer.cpp

```diff
@@ -29,18 +29,63 @@ StackRenderer::render_thread_begin(PyThreadState* tstate,
         return;
     }
 
+    ddup_push_threadinfo(sample, static_cast<int64_t>(thread_id), static_cast<int64_t>(native_id), name);
+    ddup_push_walltime(sample, 1000 * wall_time_us, 1);
+
     // Get the current time in ns in a way compatible with python's time.monotonic_ns(), which is backed by
     // clock_gettime(CLOCK_MONOTONIC) on linux and mach_absolute_time() on macOS.
     // This is not the same as std::chrono::steady_clock, which is backed by clock_gettime(CLOCK_MONOTONIC_RAW)
     // (although this is underspecified in the standard)
+    int64_t now_ns = 0;
     timespec ts;
     if (clock_gettime(CLOCK_MONOTONIC, &ts) == 0) {
-        auto now_ns = static_cast<int64_t>(ts.tv_sec) * 1'000'000'000LL + static_cast<int64_t>(ts.tv_nsec);
+        now_ns = static_cast<int64_t>(ts.tv_sec) * 1'000'000'000LL + static_cast<int64_t>(ts.tv_nsec);
         ddup_push_monotonic_ns(sample, now_ns);
     }
 
     ddup_push_threadinfo(sample, static_cast<int64_t>(thread_id), static_cast<int64_t>(native_id), name);
     ddup_push_walltime(sample, 1000LL * wall_time_us, 1);
+
+    // We stash the current thread information in the StackRenderer instance, since a task may be begun
+    // after ending the sample.
+    stashed_thread_id = thread_id;
+    stashed_native_id = native_id;
+    stashed_thread_name = std::string(name);
+    stashed_wall_time_us = wall_time_us;
+    stashed_end_time_ns = now_ns;
+}
+
+void
+StackRenderer::render_task_begin(std::string_view name)
+{
+    static bool failed = false;
```

Comment:
It should be global, since there's currently no way for us to deduce the scope of the problem.  If we fail to get a sample, we stop creating samples forever.

That said, I see there's actually a split here.  We should probably bind the error conditions for both the task and the thread begin sections.